assessment/collectors/app_code_collector.go

```diff
@@ -431,38 +461,56 @@ func (m *MigrationCodeSummarizer) AnalyzeFile(ctx context.Context, projectPath,
 		llmResponse = m.parseJSONWithRetries(m.geminiFlashModel, prompt, llmResponse, "analyze-non-dao-class-"+filepath)
 		isDataAccessObject = false
 
-		methodSignatures, err := m.extractPublicMethodSignatures(llmResponse)
-		if err != nil {
-			logger.Log.Error("Error extracting method signatures from Non-DAO analysis response: ", zap.Error(err))
-		} else {
-			extractedMethodSignatures = methodSignatures
+		if llmResponse != "" {
+			methodSignatures, err := m.extractPublicMethodSignatures(llmResponse)
+			if err != nil {
+				logger.Log.Error("Error extracting method signatures from Non-DAO analysis response: ", zap.Error(err))
+			} else {
+				extractedMethodSignatures = methodSignatures
+			}
 		}
+
 	}
 	logger.Log.Debug("File Analysis LLM Response: ", zap.String("response", llmResponse))
 
-	codeAssessment, err := parser.ParseFileAnalyzerResponse(projectPath, filepath, llmResponse, isDataAccessObject, fileIndex)
+	codeAssessment, queryResults, err := parser.ParseFileAnalyzerResponse(projectPath, filepath, llmResponse, isDataAccessObject, fileIndex)
 
 	if err != nil {
-		return &FileAnalysisResponse{emptyAssessment, extractedMethodSignatures, projectPath, filepath}
+		return &FileAnalysisResponse{emptyAssessment, extractedMethodSignatures, projectPath, filepath, queryResults}
 	}
 
-	return &FileAnalysisResponse{codeAssessment, extractedMethodSignatures, projectPath, filepath}
+	return &FileAnalysisResponse{codeAssessment, extractedMethodSignatures, projectPath, filepath, queryResults}
 }
 
 func (m *MigrationCodeSummarizer) extractPublicMethodSignatures(fileAnalysisResponse string) ([]any, error) {
 	var responseMap map[string]any
+
 	err := json.Unmarshal([]byte(fileAnalysisResponse), &responseMap)
 	if err != nil {
+		fmt.Println("llmresponse: ", fileAnalysisResponse)
```

Comment:
please remove the print statements and use loggers where needed

Reply:
Thanks for pointing out, I had added these for debugging, removed